File: frontend/templates/react_clean_architecture/src/domain/model/test/template-item.test.ts
import { describe, expect, it } from "vitest";

import { TemplateItemId, TemplateItemTitle } from "../src";

describe("TemplateItem values", () => {
  it("trims surrounding whitespace", () => {
    expect(new TemplateItemId(" item-id ").value).toBe("item-id");
    expect(new TemplateItemTitle(" Dummy item ").value).toBe("Dummy item");
  });

  it("rejects empty text", () => {
    expect(() => new TemplateItemId(" ")).toThrow();
    expect(() => new TemplateItemTitle(" ")).toThrow();
  });
});

File: mobile/templates/riverpod_clean_architecture/lib/domain/model/test/template_item_test.dart
import 'package:model/model.dart';
import 'package:test/test.dart';

void main() {
  test('item values trim surrounding whitespace', () {
    final item = TemplateItem(
      id: TemplateItemId(' item-id '),
      title: TemplateItemTitle(' Dummy item '),
    );

    expect(item.id.value, 'item-id');
    expect(item.title.value, 'Dummy item');
  });

  test('item values reject empty text', () {
    expect(() => TemplateItemId(' '), throwsArgumentError);
    expect(() => TemplateItemTitle(' '), throwsArgumentError);
  });
}
